### PR TITLE
Upgrading xcode to v12

### PR DESCRIPTION
**Pull request by shreyas098**
### Ticket Link: 

https://jira.crossroads.org.hk/browse/GCW-3608

### What does this PR do?

Upgrades xcode version to 12 and sdk version (automatically upgraded to 14 because of code upgrade)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -320,7 +320,7 @@ jobs:
   www_deploy:
     <<: *defaults
     docker:
-      - image: circleci/ruby:2.6.0-node
+      - image: circleci/ruby:2.7.1
```
Comment (steveyken) [inline .circleci/config.yml]: Ditto other comments re trying `cimg/ruby:2.7.3`

File: .ruby-version [hunk 2]
```diff
@@ -1 +1 @@
-ruby-2.5.5
+ruby-2.7.1
```
Comment (steveyken) [inline .ruby-version]:
```suggestion
ruby-2.7.3
```

**Review by steveyken — COMMENTED** [review]
Does cordova-ios need an upgrade? Need to check app builds before merging to master

File: .circleci/config.yml (hunk 1 above)
Comment (shreyas098) [reply .circleci/config.yml]: done

File: .ruby-version (hunk 2 above)
Comment (shreyas098) [reply .ruby-version]: done

File: .circleci/config.yml [hunk 3]
```diff
@@ -320,7 +320,7 @@ jobs:
   www_deploy:
     <<: *defaults
     docker:
-      - image: circleci/ruby:2.6.0-node
+      - image: circleci/ruby:2.7.3
```
Comment (steveyken) [inline .circleci/config.yml]:
```suggestion
      - image: cimg/ruby:2.7.3
```

This is the newer set of images... note the name change from circleci to cimg https://circleci.com/developer/images/image/cimg/ruby

**Review by swatijadhav — APPROVED** [review]

**Pull request merged**